driver-sync/src/test/functional/com/mongodb/client/RetryableWritesProseTest.java

```diff
@@ -0,0 +1,205 @@
+        // This will ensure that initial discovery takes place.
+        MongoDatabase database = clientUnderTest.getDatabase(databaseName);
+        MongoCollection<Document> collection = database.getCollection(collectionName);
+        Document doc = new Document("_id", 1).append("x", 11);
+        try {
+            System.out.println("--- inserting initial document...");
+            collection.insertOne(doc);
+            System.out.println("--- DONE inserting initial document");
+        } catch (MongoServerException ex) {
+            fail("Initial insert failed");
+        }
+
+        // Using the fail point client, activate the fail point by setting mode to "alwaysOn".
+        activateFailPoint();
+
+        // Using the step down client, step down the primary by executing the command { replSetStepDown: 60, force: true}.
+        // This operation will hang so long as the fail point is activated. When the fail point is later deactivated,
+        // the step down will complete and the primary's client connections will be dropped. At that point, any ensuing
+        // network error should be ignored.
+        MongoDatabase stepDownDB = stepDownPrimary();
+
+
+        System.out.println("Waiting for secondary...");
+        try {
+            while (!stepDownDB.runCommand(new BsonDocument("isMaster", new BsonInt32(1))).getBoolean("secondary")) {
+                Thread.sleep(1000);
+            }
+        } catch (InterruptedException ex) {
+        }
+        System.out.println("--- DONE stepping down primary");
+
+        // Using the client under test, insert a document and observe a successful write result. The test MUST assert
+        // that the insert command fails once against the stepped down node and is successfully retried on the newly
+        // elected primary (after SDAM discovers the topology change). The test MAY use APM or another means to observe
+        // both attempts.
+        try {
+            System.out.println("--- insert one doc, should fail...");
+            collection.insertOne(new Document("_id", 2).append("x", 22));
+            fail("Exception should have been raised on insert");
+        } catch (MongoException ex) {
+            System.out.println("Expected exception raised: " + ex.getMessage());
+        }
+
+        try {
+            System.out.println("--- insert one doc, should pass...");
+            collection.insertOne(new Document("_id", 2).append("x", 22));
+            System.out.println("--- DONE inserting one doc");
+        } catch (MongoException ex) {
+            fail("Exception should not have been raised on insert: " + ex.getMessage());
+        }
+    }
+
+    private boolean canRunTests() {
+        return serverVersionAtLeast(3, 6) && isDiscoverableReplicaSet();
+    }
+
+    private void setUpClientUnderTest() {
+        MongoClientSettings settings = getMongoClientSettingsBuilder()
```

Comment:
Is this the proper way of setting the heartbeat frequency? I only found one example of this, and the code below is what was shown.